core/io/http_client.cpp

```diff
@@ -226,6 +226,7 @@ Error HTTPClient::request(Method p_method, const String &p_url, const Vector<Str
 	}
 
 	status = STATUS_REQUESTING;
+	head_request = p_method == METHOD_HEAD;
```

Comment:
This should be done in `HTTPClient::request_raw` too.
Also needs initialization in constructor and `HTTPClient::close`.